Create bezier

prajwalprabhu opened the pull request:
Returns the control points  if you pass knot(end) points of bezier. the control points will form an smooth cubic bezier curve

ctrlcctrlv commented:
Woah, what's with the big patch to `src/glyphbuilder/mod.rs`? Is this just `rustfmt`? That'd be dangerous to merge if it will change existing behavior.

prajwalprabhu commented:
> Woah, what's with the big patch to `src/glyphbuilder/mod.rs`? Is this just `rustfmt`? That'd be dangerous to merge if it will change existing behavior.

ya it is just rustfmt

prajwalprabhu commented:
Removed some of the unwanted commit messages

ctrlcctrlv commented on src/lib.rs:
```diff
@@ -1,38 +1,43 @@
-pub mod glyphbuilder;
-#[cfg(feature="default")]
+pub mod rect;
+pub mod create_bazier;
```
This is misspelled, but the module name is pretty confusing, I suggest you call it `fit_to_points` or something, and move the module `get_control_points` to be under `fit_to_points`.

prajwalprabhu replied:
Changed it  👍.

ctrlcctrlv commented on src/fit_to_points/get_control_points.rs:
```diff
@@ -0,0 +1,87 @@
+                (2. * knots[0].x + knots[1].x) / 3.,
+                (2. * knots[0].y + knots[1].y) / 3.,
+            ),
+            PointType::Undefined,
```
Using this point type is dangerous. It only exists for use in `GlifPoint` and must not be used in `Point`'s.

prajwalprabhu replied:
So can I use a `Vector` here

ctrlcctrlv replied:
Yes.

ctrlcctrlv replied:
Or actually `PointType::Move` is OK too if they are both single floating points.

ctrlcctrlv replied:
There's also `PointType::OffCurve`, which is quite dangerous to be put in a `Contour`, but if you go over it carefully and make sure no `PointType::OffCurve` in final `Contour` it's okay.

prajwalprabhu replied:
can I use `PointType::Move` for every point.

ctrlcctrlv replied:
Basically, `Point` should never be `Undefined`, and `Contour` should only ever contain `Move`, `Line`, or `Curve` _at the time API returns_. You can use the other types while processing, that's what they're for. `Undefined` is only for `GlifPoint`.

prajwalprabhu replied:
I actually don't need a `Point` type over there I just have to return two `f32` ,So I will use `Vector` or `(f32,f32)`

ctrlcctrlv replied:
Yes, good idea.

prajwalprabhu replied:
I tried to use `Vector` but there the types are in `f64` so then I used `(f32,f32)`

ctrlcctrlv commented:
Does this code respect `PointType::Line` yet

prajwalprabhu commented:
Ya it respects `PointType::Line` ,we don't need a control point for it,So it only calculates control points for `PointType::Curve`

ctrlcctrlv commented:
Thanks for your contribution…what name should go in AUTHORS file? `T. Prajwal Prabhu`? `T` stand for anything, or would you prefer it just to be `T.`? :)

ctrlcctrlv commented:
(Sorry for my ignorance, but if `T.` has a meaning similar to Dr/Ms/… then please tell me that too because it shouldn't be there in that case)

prajwalprabhu commented:
> (Sorry for my ignorance, but if `T.` has a meaning similar to Dr/Ms/… then please tell me that too because it shouldn't be there in that case)

T is part of my name which refers to my family.

ctrlcctrlv commented:
Thanks :) I merged this, fixed the merge conflict in #21, merged #21, and added you to AUTHORS/README in https://github.com/MFEK/math.rlib/commit/08b002c939f2167e27e724b84e455c4e49855993. Hope to see more useful work from you :D when will we have simplify do you think?

prajwalprabhu commented:
> Thanks :) I merged this, fixed the merge conflict in #21, merged #21, and added you to AUTHORS/README in [08b002c](https://github.com/MFEK/math.rlib/commit/08b002c939f2167e27e724b84e455c4e49855993). Hope to see more useful work from you :D when will we have simplify do you think?

Thank you for the merge.
Regarding simplify , 
We have to make decision whether we could remove the knot point or not ,So I'm still struggling in that part . Do you have any idea about it ?(as you have contributed to fontforge)

Then
we will have :
*starting point
*ending point
*old control points
*and slope of the line on which the new CP should be ,
*and we could use the old end point as a point on the curve
Then we could go with kind off trial and error ,we will increase or decrease the y on line joining CP and endpoint ,and we will get the curve and we will check whether the point exist or not on the curve with that CP.
